client/my-sites/guided-transfer/host-credentials-page/index.jsx

```diff
@@ -29,16 +40,34 @@ class HostCredentialsPage extends Component {
 		this.setFieldValue( fieldName, e.target.value );
 	}
 
+	redirectToCart = () => {
+		upgradesActions.addItem( cartItems.guidedTransferItem() );
+		page.redirect( `/checkout/${ this.props.siteSlug }` );
+	}
+
 	submit = () => {
-		// Future PR: This function will trigger the Redux API call
+		const payload = {
+			...this.state.fieldValues,
+			host_slug: this.props.hostSlug
+		};
+
+		this.props.submit( payload )
+			.then( didSubmit => {
```

Comment:
In the early stages of this PR, I'd experimented with a `Redirect` component, but it caused weird behaviour when pressing the Back button. For now I've solved this one with ed6c7dd which just checks for updated props to trigger the redirect.